packages/headless/src/features/facets/range-facets/date-facet-set/date-facet-selectors.ts

```diff
@@ -16,10 +17,17 @@ function isDateFacetResponse(
 export const dateFacetResponseSelector = (
   state: SearchSection & DateFacetSection,
   facetId: string
-) => {
+): DateFacetResponse | undefined => {
   const response = baseFacetResponseSelector(state, facetId);
+  const request = state.dateFacetSet[facetId];
+
   if (isDateFacetResponse(state, response)) {
-    return response;
+    return {
```

Comment:
Instead of doing the mapping (from request RelativeDate to response relative dates) in the selector & controller, we could also do it in the search slice directly.

Reply:
No other kind of postProcess is done in the slice for the response, wouldn't look out of place

```typescript
function handleFulfilledSearch(
  state: SearchState,
  action: ReturnType<SearchAction['fulfilled']>
) {
  state.error = null;
  state.response = doSomethingWithTheDateFacet(action.payload.response);
  state.queryExecuted = action.payload.queryExecuted;
  state.duration = action.payload.duration;
  state.isLoading = false;
}
```

Controllers like the DateFilter use the DateFacetSelector which maps the RelativeDate.

Reply:
Forgot to mention that the SearchState has no access to the DateFacetSetState because they are different reducers, so my initial comment was a bit incorrect 😅